ui/app/utils/openapi-helpers.ts

```diff
@@ -0,0 +1,129 @@
+import { dasherize } from '@ember/string';
+import { singularize } from 'ember-inflector';
+
+// TODO: Consolidate with openapi-to-attrs once it's typescript
+
+interface Path {
+  path: string;
+  itemType: string;
+  itemName: string;
+  operations: string[];
+  action: string;
+  navigation: boolean;
+  param: string | false;
+}
+interface PathsInfo {
+  apiPath: string;
+  itemType: string;
+  itemTypes: string[];
+  paths: Path[];
+}
+
+interface OpenApiParameter {
+  description?: string;
+  in: string;
+  name: string;
+  required: boolean;
+  schema: object;
+}
+interface DisplayAttrs {
+  itemType: string;
+  action: string;
+  navigation?: boolean;
+  description?: string;
+  name?: string;
+  group?: string;
+  value?: string | number;
+  sensitive?: boolean;
+}
+interface OpenApiAction {
+  parameters: Array<{ name: string }>;
+}
+interface OpenApiPath {
+  description?: string;
+  parameters: OpenApiParameter[];
+  'x-vault-displayAttrs': DisplayAttrs;
+  get?: OpenApiAction;
+  post?: OpenApiAction;
+  delete?: OpenApiAction;
+}
+
+// Take object entries from the OpenAPI response and consolidate them an object which includes itemTypes, operations, and paths
```

Comment:
not worth another commit, but small typo
```suggestion
// Take object entries from the OpenAPI response and consolidate them in/to (?) an object which includes itemTypes, operations, and paths
```